### PR TITLE
fix: only pass output to sonar when available (RE-2814)

### PR DESCRIPTION
**Pull request by erikburt**
### Changes

Fix sonarqube occasionally reporting 0% coverage when core tests are not run, and therefore there's no test or coverage output.

### Testing

Tested in #13866 - https://github.com/smartcontractkit/chainlink/actions/runs/9961379279/job/27522822549.

### Requires Dependencies

N/A

### Resolves Dependencies

N/A

---

[RE-2814](https://smartcontract-it.atlassian.net/browse/RE-2814)

[RE-2814]: https://smartcontract-it.atlassian.net/browse/RE-2814?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by chainchad — APPROVED** [review]
👍 👍

File: .github/workflows/ci-core.yml [hunk 1]
```diff
@@ -340,23 +340,50 @@ jobs:
...
           echo "sonarqube_coverage_report_paths=$(find go_core_tests_logs -name coverage.txt | paste -sd "," -)" >> $GITHUB_OUTPUT
+          echo "sonarqube_lint_report_paths=$(find golangci-lint-report -name golangci-lint-report.xml | paste -sd "," -)" >> $GITHUB_OUTPUT
+
+      - name: Check SonarQube Report Paths
+        id: check_sonarqube_paths
+        run: |
+          ARGS=""
+
+          if [[ -z "${{ steps.sonarqube_report_paths.outputs.sonarqube_tests_report_paths }}" ]]; then
+            echo "::warning::No test report paths found, will not pass to sonarqube"
```
Comment (HenryNguyen5) [inline .github/workflows/ci-core.yml:358]: nice, good usage of gha output commands

File: .github/workflows/ci-core.yml [hunk 2]
```diff
@@ -340,23 +340,50 @@ jobs:
...
+            ARGS="$ARGS -Dsonar.go.coverage.reportPaths=${{ steps.sonarqube_report_paths.outputs.sonarqube_coverage_report_paths }}"
+          fi
+
+          if [[ -z "${{ steps.sonarqube_report_paths.outputs.sonarqube_lint_report_paths }}" ]]; then
+            echo "::warning::No lint report paths found, will not pass to sonarqube"
+          else
+            ARGS="$ARGS -Dsonar.go.golangci-lint.reportPaths=${{ steps.sonarqube_report_paths.outputs.sonarqube_lint_report_paths }}"
+          fi
+
+          echo "SONARQUBE_ARGS=$ARGS" >> $GITHUB_ENV
```
Comment (HenryNguyen5) [inline .github/workflows/ci-core.yml:375]: We shouldn't have any multiline output problems to deal with, right?

File: .github/workflows/ci-core.yml [hunk 3]
```diff
@@ -340,23 +340,50 @@ jobs:
...
         id: sonarqube_report_paths
         shell: bash
         run: |
           echo "sonarqube_tests_report_paths=$(find go_core_tests_logs -name output.txt | paste -sd "," -)" >> $GITHUB_OUTPUT
           echo "sonarqube_coverage_report_paths=$(find go_core_tests_logs -name coverage.txt | paste -sd "," -)" >> $GITHUB_OUTPUT
+          echo "sonarqube_lint_report_paths=$(find golangci-lint-report -name golangci-lint-report.xml | paste -sd "," -)" >> $GITHUB_OUTPUT
+
+      - name: Check SonarQube Report Paths
+        id: check_sonarqube_paths
+        run: |
```
Comment (HenryNguyen5) [inline .github/workflows/ci-core.yml:354]: wdyt about combining this into 1 step? it'd improve the readability by making the var paths significantly shorter

**Review by HenryNguyen5 — APPROVED** [review]
couple nits, looks good otherwise

File: .github/workflows/ci-core.yml (hunk 2 above)
Comment (erikburt) [reply .github/workflows/ci-core.yml:375]: I don't think so. This is a standard practice amongst many repos utilizing sonar scan.

File: .github/workflows/ci-core.yml (hunk 3 above)
Comment (erikburt) [reply .github/workflows/ci-core.yml:354]: Will complete in follow-up PR.

**Pull request merged**